condition/scenario_conditions/include/scenario_conditions/condition_base.h

```diff
@@ -15,8 +17,11 @@ class ConditionBase
 
   ConditionBase(const std::string & type)
     : type_ { type }
-    , name_ { type + "Condition<" + boost::lexical_cast<std::string>(this) + ">" }
-  {}
+  {
+    std::stringstream ss;
+    ss << type << "Condition<" << static_cast<const void*>(this) << ">";
```

Comment:
:+1: for removing the boost dep. This seems odd to save the pointer as a string in `name_`. But whatever..